Price formatter

bthaile opened the pull request:
## Related Issue

Closes #454

Adds formatting price to create a subscript 

## Summary of Changes

Addes helper to format prices,

## Need Regression Testing


- [ ] Yes
- [x] No

## Risk Assessment

- [x] Low
- [ ] Medium
- [ ] High

## Additional Notes

This only a UI component with helper method and tests. It's not been integrated into the existing UI.

## Screenshots (if applicable)
![Screenshot 2025-02-04 at 18 09 07](https://github.com/user-attachments/assets/899a84d2-b71e-4dec-a94d-cffeb1de876b)

![Screenshot 2025-02-04 at 18 09 21](https://github.com/user-attachments/assets/2aa21b4c-a75a-4e92-a54c-1b2e41fa5afb)

bthaile committed:
add price formatter helper

bthaile committed:
add token price UI component

tombeckenham reviewed (approved):
Good stuff. Loads of tests. Note if you rebase to dev, the playwright tests should pass. There was an issue I introduced late in the piece that I fixed in 2.7.1.

tombeckenham commented:
Sorry, you need to merge to dev not master

tombeckenham reviewed (changes requested):
You need to merge to dev. You'll have to recreate this Pr.

bthaile commented:
ok, updated destination branch and added some screenshots. Seems there are a few places to standardize displaying token price, so I'll go ahead and use the new component. 

Initially I was just going to make the component and allow the UI team to update viz, but I can go ahead and do it.

Kay-Zee commented:
does it make sense just to add the component first, and then make use of it in a different PR?

Kay-Zee commented on src/shared/utils/formatPrice.test.ts:
```diff
@@ -0,0 +1,275 @@
+  it('should condense numbers with many zeros after decimal', () => {
+    const testCases = [
+      {
+        input: 0.0001234, // 4 zeros total, should not condense yet
```
3 zeros after the decimal

Kay-Zee commented on src/shared/utils/formatPrice.test.ts:
```diff
@@ -0,0 +1,275 @@
+        },
+      },
+      {
+        input: 0.00001234, // 4 zeros total, should condense with zeroPart 4
```
4 zeros after the decimal, condense with zeroPart 3

Kay-Zee commented on src/shared/utils/formatPrice.test.ts:
```diff
@@ -0,0 +1,275 @@
+import { formatPrice } from './formatPrice';
+
+describe('formatPrice', () => {
+  const DEFAULT_THRESHOLD = 4;
```
Can we add a quick blurb about the threshold.

// Number of zeros after the decimal that causes format to be condensed.
// i.e. condense if # zeros after decimal >= threshold.

Kay-Zee commented on src/shared/utils/formatPrice.test.ts:
```diff
@@ -0,0 +1,275 @@
+    });
+  });
+
+  it('should handle sequential test cases for all zero counts', () => {
```
whats the difference between this test and the one above `should handle numbers with up to 8 zeros after decimal`? If you just added one more case here, would it not cover the above?

Kay-Zee commented on src/shared/utils/formatPrice.test.ts:
```diff
@@ -0,0 +1,275 @@
+  it('should respect different threshold values', () => {
+    const testCases = [
+      {
+        input: 0.0001234, // 4 zeros
```
3 zeros?

Kay-Zee commented on src/shared/utils/formatPrice.test.ts:
```diff
@@ -0,0 +1,275 @@
+        },
+      },
+      {
+        input: 0.0001234, // 4 zeros
```
3 zeros? or just remove the comment

tombeckenham reviewed (changes requested):
Good points from KayZee. If you resolve those I'm happy

bthaile commented:
Closing this PR, since I originally branched off master. Took comments in consideration.

Created new branch off dev Here is the PR
https://github.com/onflow/FRW-Extension/pull/465
